src/components/scenes/Loans/ManageCollateralScene.tsx

```diff
@@ -168,87 +151,91 @@ export const ManageCollateralScene = <T extends keyof ParamList>(props: Props<T>
-            nativeAmount: actionNativeCryptoAmount,
-            ...(selectedAsset.paymentMethod != null ? { paymentMethodId: selectedAsset.paymentMethod.id } : {}),
-            ...(selectedAsset.tokenId != null ? { tokenId: selectedAsset.tokenId } : {})
-          }
-          setActionOp(
-            await makeAaveBorrowAction({
+          break
+        case 'loan-borrow':
+          {
+            const destination: LoanAsset = {
+              wallet: borrowEngineWallet,
+              tokenId: selectedAsset.tokenId,
+              nativeAmount: actionNativeCryptoAmount,
+              ...(selectedAsset.paymentMethod != null ? { paymentMethodId: selectedAsset.paymentMethod.id } : {}),
+              ...(selectedAsset.tokenId != null ? { tokenId: selectedAsset.tokenId } : {})
+            }
+            actionOp = await makeAaveBorrowAction({
               borrowEngineWallet,
               borrowPluginId,
               destination
             })
-          )
-        }
-        break
-      default: {
-        const actionOp: ActionOp = {
-          type: 'seq',
-          actions: [
-            {
-              type: actionOpType,
-              borrowPluginId,
-              nativeAmount: actionNativeCryptoAmount,
-              walletId: borrowEngineWallet.id,
-              tokenId: selectedAsset.tokenId
-            }
-          ]
+          }
+          break
+        default: {
+          actionOp = {
+            type: 'seq',
+            actions: [
+              {
+                type: actionOpType,
+                borrowPluginId,
+                nativeAmount: actionNativeCryptoAmount,
+                walletId: borrowEngineWallet.id,
+                tokenId: selectedAsset.tokenId
+              }
+            ]
+          }
         }
-        setActionOp(actionOp)
       }
+      setActionProgram(await makeActionProgram(actionOp))
     }
 
     return () => {}
-  }, [actionNativeCryptoAmount, actionOpType, borrowEngineWallet, borrowPluginId, selectedAsset])
+  }, [actionNativeCryptoAmount, actionOpType, borrowEngineWallet, borrowPluginId, isLtvExceeded, selectedAsset])
 
-  // @ts-expect-error
+  // Calculate Network Fees
   useAsyncEffect(async () => {
-    if (zeroString(actionNativeCryptoAmount) || isLtvExceeded) {
-      setApprovalAction(null)
-      return
+    if (actionProgram != null) {
+      // Get fees
+      const actionProgramState = {
+        clientId,
+        programId: actionProgram.programId,
+        effective: false,
+        executing: false,
+        lastExecutionTime: 0,
+        nextExecutionTime: 0
+      }
+      const executionContext = { account, clientId }
+      const executionOutputs = await dryrunActionProgram(executionContext, actionProgram, actionProgramState, false)
+      const networkFeeMap = getExecutionNetworkFees(executionOutputs)
+      setFeeNativeAmount(networkFeeMap[borrowEngineWallet.currencyInfo.currencyCode]?.nativeAmount ?? '0')
```

Comment:
We should store the entire `NetworkFeeMap` as state and then derive the `feeNativeAmount` from that state. This way when we go to fix our fee calculation to include all network fee amounts, we would only need to remove the derived value and directly pass the `NetworkFeeMap` object to the component/function handling the calculation of all fees.